### PR TITLE
Allow absolute paths in deps.edn :local/root

### PR DESCRIPTION
**Pull request by Cnly**
Previously if I have something like

```clj
clj-kondo/clj-kondo {:local/root "/Users/me/git/clj-kondo"}
```

as a dependency in `deps.edn`, lsp will fail with

```
java.lang.IllegalArgumentException: /Users/me/git/clj-kondo is not a relative path
```

This PR fixes that issue.

### CODE REVIEW

**Comment by ericdallo** [comment]
@Cnly d you think it's related with this issue?
https://clojurians.slack.com/archives/CPABC1H61/p1639747789213400

File: src/clojure_lsp/source_paths.clj [hunk 1]
```diff
@@ -80,7 +80,8 @@
                            (remove nil?))
           deps-files-by-local-root (->> local-roots
                                         (map (fn [local-root]
-                                               (let [sub-deps-file (io/file root-path local-root "deps.edn")]
+                                               (let [local-root (shared/relativize-filepath local-root root-path)
+                                                     sub-deps-file (io/file root-path local-root "deps.edn")]
```
Comment (ericdallo) [inline src/clojure_lsp/source_paths.clj:84]: Could you create a test for that?

Comment (ericdallo) [reply src/clojure_lsp/source_paths.clj:84]: seems like CI is failing because of this change

Comment (Cnly) [reply src/clojure_lsp/source_paths.clj:84]: Right, should be fixed now with a new testcase.

**Review by ericdallo — APPROVED** [review]

**Comment by Cnly** [comment]
Should I also add a changelog entry for this one?

**Comment by ericdallo** [comment]
Thank you very much for the help!

**Pull request merged**